extension/extensionauth/extensionauthtest/error_client.go

```diff
@@ -4,25 +4,48 @@
 package extensionauthtest // import "go.opentelemetry.io/collector/extension/extensionauth/extensionauthtest"
 
 import (
+	"context"
 	"errors"
 	"net/http"
 
 	"google.golang.org/grpc/credentials"
 
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/extension"
 	"go.opentelemetry.io/collector/extension/extensionauth"
 )
 
 var errMockError = errors.New("mock Error")
 
-// NewErrorClient returns a new extensionauth.Client that always returns an error on any of the client methods.
-func NewErrorClient(opts ...extensionauth.ClientOption) (extensionauth.Client, error) {
-	errorOpts := []extensionauth.ClientOption{
-		extensionauth.WithClientRoundTripper(func(http.RoundTripper) (http.RoundTripper, error) {
-			return nil, errMockError
-		}),
-		extensionauth.WithClientPerRPCCredentials(func() (credentials.PerRPCCredentials, error) {
-			return nil, errMockError
-		}),
-	}
-	return extensionauth.NewClient(append(errorOpts, opts...)...)
+var (
+	_ extension.Extension      = (*errorClient)(nil)
+	_ extensionauth.HTTPClient = (*errorClient)(nil)
+	_ extensionauth.GRPCClient = (*errorClient)(nil)
+)
+
+type errorClient struct{}
```

Comment:
```suggestion
type errorClient struct{
  component.StartFunc
  component.ShutdownFunc
}
```

Reply:
For me the rule when a Func like "StartFunc" should implement the interface function "Start" is that, when we have an unsealed interface like Component, users can do what I suggested and not have to write the signature themselves. And when is sealed like in the Factory interface we don't need them since we control the only implementation.

Reply:
So in this case we should provide the functions for the http/grpc clients and they should implement the signature, so that you can write this as:


```
type errorClient struct{
  component.StartFunc
  component.ShutdownFunc
  extensionauth.ClientRoundTripperFunc
}

func NewErrorClient(err error) extension.Extension {
  return &errorClient{ClientRoundTripperFunc: func() (credentials.PerRPCCredentials, error) { return err; }}
}
```